core/src/main/scala/chisel3/properties/Object.scala

```diff
@@ -44,6 +57,36 @@ class DynamicObject private[chisel3] (val className: ClassType) extends HasId wi
   }
 }
 
+object DynamicObject {
```

Comment:
Now that it's possible for users to create these instances, we should include ScalaDoc with an example on `class DynamicObject`